drv/update-api/src/lib.rs

```diff
@@ -53,6 +54,22 @@ pub struct ImageVersion {
     pub version: u32,
 }
 
+#[derive(
+    Copy,
+    Clone,
+    FromBytes,
+    AsBytes,
+    PartialEq,
+    Eq,
+    Serialize,
+    Deserialize,
+    SerializedSize,
+)]
+#[repr(C)]
+pub struct UpdateStatus {
+    pub active_slot: u8,
+}
```

Comment:
This looks like it is being sent back and forth between RoT and SP. This also seems like a structure that could change as we discover new data we need to send back and forth. Do we need a structure version on this to be able to handle changes in fields?

Reply:
That's a great question @labbott. Perhaps. This structure itself will be embedded in a sprot structure before actually being sent over the wire, and so maybe that's where the versioning should happen. There's already a header with a [Protocol field](https://github.com/oxidecomputer/hubris/blob/master/drv/sprot-api/src/lib.rs#L260-L268), but that's just a u8 and really is meant to version the whole sprot API. Maybe there should be a separate per message version? I dealt with this in [sprockets](https://github.com/oxidecomputer/sprockets/blob/main/common/src/msgs.rs#L24-L36) by versioning the whole interface, but sprockets is expected to change much less than other things, and we want to change the whole protocol atomically.

It's certainly easy enough to add a version number here, and I'm open to it. @lzrd @jgallagher what do you think?

Reply:
It needs to be stabilized. There is a version on the protocol as a whole which must imply a consistent set of operations used to manage firmware updates (if nothing else).
We could add to the message set at any time while old implementations would return "not implemented".
Andrew is making some of these commits when they are still works in progress.